astropy/time/formats.py

```diff
@@ -1241,6 +1297,25 @@ def parse_string(self, timestr, subfmts):
                              .format(timestr, self.name))
 
     def set_jds(self, val1, val2):
+        """Parse the time strings contained in val1 and set jd1, jd2"""
+        # If specific input subformat is required then use the Python parser.
+        # Also do this if Time format class does not define `use_fast_parser`
+        # or if the fast parser is entirely disabled.
+        if (self.in_subfmt != '*'
+                or not self.__class__.__dict__.get('use_fast_parser')
+                or conf.use_fast_parser == 'False'):
+            self.set_jds_python(val1, val2)
```

Comment:
And then obviously assign to `self.jd1, self.jd2` here and below. Advantage is that it separates the tasks of calculating jd1 and jd2 from setting.